addons/stock/models/stock_move_line.py

```diff
@@ -219,6 +219,9 @@ def write(self, vals):
         if self.env.context.get('bypass_reservation_update'):
             return super(StockMoveLine, self).write(vals)
 
+        if 'product_id' in vals and any(ml.state == 'done' and vals['product_id'] != ml.product_id.id for ml in self):
+            raise UserError('You are not allowed to modify the product on an operation done.')
```

Comment:
also problematic is assigned and partially available.. and maybe the source of "the cannot unreserve more than.." tickets
if a sml is reserved and the product is changed trough an import, the original quant stays reserved and some of the newly changed producs are unreserved

cc @nvn-odoo @jev-odoo

Reply:
@sle-odoo TY for the info ;)

Reply:
i would change the message to
"Changing the product of this record is forbidden at this point."

Reply:
If you suggest changing the message, then the 'at this point' seems not very precise for the customer.

It could be "Changing the product of this record is forbidden on state 'Done'." or something like that. The more precise info for the customer, the less Support ticket ;)

Reply:
@sle-odoo indeed, thank you for the notification :-)